crates/matrix-sdk-ui/src/timeline/controller/observable_items.rs

```diff
@@ -553,6 +553,7 @@ impl<'e> ObservableItemsTransactionIterBuilder<'e> {
     }
 
     /// Build the iterator.
+    #[allow(clippy::iter_skip_zero)]
```

Comment:
Would not be needed if you like my suggestion, but maybe you like the regularity of always skipping something.

Reply:
Oh wait, you did it so you can return a concrete type! I guess I am ok with that then...

A `Box<dyn Iterator>` would be simpler, but there is a cost...

Reply:
There is a cost indeed. Also, not all `Iterable` implements `ExactSizeIterator` nor `DoubleEndedIterator`, which is a requirement. We could use `Box<dyn Iterator + ExactSizeIterator + DoubleEndedIterator>` but I'm not sure I want this for the moment. We don't need _this power_ for the moment, a concrete type is fine.